doc/dev/test_proxy_migration_guide.md

```diff
@@ -0,0 +1,174 @@
+# Guide for migrating to the test proxy from vcrpy
+
+This guide describes the changes that service SDKs should make to their test frameworks in order to take advantage of
+the Azure SDK test proxy.
+
+Documentation of the motivations and goals of the test proxy can be found [here][general_docs] in the azure-sdk-tools
+GitHub repository, and documentation of how to set up and use the proxy can be found [here][detailed_docs].
+
+## Update existing tests
+
+### Current test structure
+
+Test classes currently inherit from AzureTestCase, and test methods can optionally use decorators:
+
+```py
+from devtools_testutils import AzureTestCase
+
+class TestExample(AzureTestCase):
+
+    def test_example(self):
+        ...
+
+    @ExamplePreparer()
+    def test_example_with_preparer(self):
+        ...
+```
+
+### New test structure
+
+To use the proxy, test classes should inherit from AzureRecordedTestCase and recorded test methods should use a
+RecordedByProxy decorator:
+
+```py
+from devtools_testutils import AzureRecordedTestCase, RecordedByProxy
+
+class TestExample(AzureRecordedTestCase):
+
+    @RecordedByProxy
+    def test_example(self):
+        ...
+
+    @ExamplePreparer()
+    @RecordedByProxy
+    def test_example_with_preparer(self):
+        ...
+```
+
+For async tests, import the RecordedByProxyAsync decorator from `devtools_testutils.aio` and use it in the same
+way as RecordedByProxy.
+
+> **Note:** since AzureRecordedTestCase doesn't inherit from `unittest.TestCase`, test class names need to start
+> with "Test" in order to be properly collected by pytest by default. For more information, please refer to
+> [pytest's documentation][pytest_collection].
+
+## Run the tests
+
+### Perform one-time setup
+
+The test proxy is made available for your tests via a Docker container. Run the following command to build this
+container, providing the full path to the root of your local `azure-sdk-for-python` repo in place of `{path}`:
+
+```cmd
+docker container create -v C:\\{path}\\azure-sdk-for-python:/etc/testproxy -p 5001:5001 -p 5000:5000 --name ambitious_azsdk_test_proxy azsdkengsys.azurecr.io/engsys/testproxy-lin
```

Comment:
Do we want to just point them to the `docker-start-proxy.ps1` under `eng/common/testproxy`?

It does pretty much exactly this (and figures out where the root of the repo is for you).